### PR TITLE
Add rubocop to actions

### PR DESCRIPTION
**Pull request by girijaasoni**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require:
+  - rubocop-performance
```
Comment (nofaralfasi) [inline .rubocop.yml:2]:
```suggestion
  - rubocop-performance
  - rubocop-minitest
```

File: Rakefile [hunk 2]
```diff
@@ -8,7 +8,15 @@ Rake::TestTask.new do |t|
   t.warning = ENV.key?('RUBY_WARNINGS')
 end
 
-task :default => :test
+task default: :test
```
Comment (nofaralfasi) [inline Rakefile:11]: See https://github.com/theforeman/hammer-cli-foreman-ansible/pull/29#discussion_r1505749999

File: Gemfile [hunk 3]
```diff
@@ -7,16 +7,18 @@ gem 'gettext', '>= 3.1.3', '< 4.0.0'
 gem 'hammer_cli_foreman', github: 'theforeman/hammer-cli-foreman', branch: 'master'
 
 group :test do
-  gem 'ci_reporter', '>= 1.6.3', '< 2.0.0', :require => false
+  gem 'ci_reporter', '>= 1.6.3', '< 2.0.0', require: false
```
Comment (nofaralfasi) [inline Gemfile:10]: See https://github.com/theforeman/hammer-cli-foreman-discovery/pull/57#issuecomment-1978541538

File: .github/workflows/unit_tests.yaml [hunk 4]
```diff
@@ -11,6 +11,10 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
   test:
     name: Tests
     uses: theforeman/actions/.github/workflows/test-gem.yml@v0
```
Comment (stejskalleos) [inline .github/workflows/unit_tests.yaml:20]: I would add `needs: rubocop` so the tests run only if Rubocop is green

**Review by stejskalleos — APPROVED** [review]

**Pull request merged**